operators/utilities.py

```diff
@@ -38,6 +38,52 @@ def select_all(context: Context):
 def deselect_all(context: Context):
     global_data.selected.clear()
 
+def select_invert(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    for e in generator:
+        e.selected = not e.selected
+
+def select_extend(context: Context):
+    sketch = context.scene.sketcher.active_sketch
+    if sketch:
+        generator = sketch.sketch_entities(context)
+    else:
+        generator = entities_3d(context)
+    
+    to_select = []
+    for e in generator:
+        if e.is_segment():
+            if e.selected:
+                if hasattr(e, 'p1'):
```

Comment:
Probably easier to just append e.connection_points()

Reply:
although it would be best if we could have a garantee that all elements have that function (probably should be implemented for the abstract class SlvsGenericEntity)

Reply:
but yes, definitely better to use `connection_points()` as `p1` and `p2` are way too specific. I would recommend using `not isinstance(e, class_defines.point) and isinstance(e, SlvsGenericEntity)`

Reply:
@amrsoll @hlorus  Maybe it is good idea to create mixin class like `SegmentEntity` insted/additional of `is_segment` method. We could add `connection_points()` and additionally create `coincident_points()` method to it?

Reply:
I have a feeling that these classes might change in the not-so-distant future and adding the extra mixin class isn't justifiable.

Reply:
copy that